woocommerce/woocommerce-google-analytics-integration
# Feature/consistency across gtag implementation

### Changes proposed in this Pull Request:

This is the culmination of a project to resolve some inconsistencies in the event data sent to Google Analytics along with putting safe guards in place to prevent issues being introduced in the future.

- [x] #230
- [x] #240 
- [x] #245 
- [x] #246

Closes #171, #185, #200, #233, #247

### Checks:

### Detailed test instructions:

1. Checkout `feature/consistency-across-gtag-implementation`
2. Manually [run unit tests](https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/245)
   - Run `composer install`
   - Run `bin/install-unit-tests.sh <db-name> <db-user> <db-pass> [db-host] latest latest`
   - Run `vendor/bin/phpunit`
   - Confirm that all tests pass successfully
3. Set up WooCommerce Google Analytics Integration on a test website
3. Ensure [Google Analytics Debugger](https://chrome.google.com/webstore/detail/google-analytics-debugger/jnkmfdileelhofjcijamephohjechhna) is installed and `ON` and that you are _not_ logged into your test site
4. Check [Preserve Log](https://developer.chrome.com/docs/devtools/console/reference/#persist) in the Chrome Developer Console
5. Walk through a typical customer flow on the front end of the website:
   - Selecting products
   - Adding different types of products to the cart
   - Removing products from the cart
   - Going through the checkout process
6. Confirm that while performing these actions that the corresponding Google Analytics events are being logged in the console and that the data is what's expected
